Add .travis.yml file

cking100 opened the pull request:
Added a travis build that ensures the code quality of the website by adding html, css and js linters to the build.

cking100 commented:
@nemesisdesign @atb00ker Please have a look and lemme know. And i think we need a configuration file for the stylelint like `.stylelintrc` or something

cking100 commented:
Okay will fix the matrix into a single build but i have a query here, the way i ran the `html_lint.py *.html` will this command be able to access the html files inside `gsoc` folder?

cking100 commented:
And instead of having 3 different configuration files why don't we have a package.json file that will have jshintConfig property, stylelint property and a similar property for html linter? Lemme know your thoughts on this @nemesisdesign

cking100 commented:
okay i will @nemesisdesign

cking100 commented:
@nemesisdesign @atb00ker I want to clarify a few things here as i might have misunderstood it and things are taking longer than usual:
1. l have to make a default configuration file for linters and fix the code so that it is according to the config file right?
2. The html linter which is to be run in travis build doesn't work with a config file how should i go about in that case?

nemesisdesign commented on .travis.yml:
```diff
@@ -0,0 +1,16 @@
+    - pip install html-linter
+    - npm install -g jshint
+    - npm install -g stylelint
+script:
```
leave blank line before this line

nemesisdesign commented on .travis.yml:
```diff
@@ -0,0 +1,16 @@
+node_js: "stable"
+before_install:
+    - npm install -g npm@latest
+install:
```
leave blank line before this line

nemesisdesign commented on .travis.yml:
```diff
@@ -0,0 +1,16 @@
+python:
+    - "3.5"
+node_js: "stable"
+before_install:
```
leave blank line before this line

nemesisdesign commented on .travis.yml:
```diff
@@ -0,0 +1,16 @@
+language: python
+python:
+    - "3.5"
```
use 2 spaces for indentation

nemesisdesign commented on .travis.yml:
```diff
@@ -0,0 +1,16 @@
+install:
+    - pip install html-linter
+    - npm install -g jshint
+    - npm install -g stylelint
```
use 2 spaces for indentation, same below

nemesisdesign commented on .travis.yml:
```diff
@@ -0,0 +1,16 @@
+    - html_lint.py gsoc/*.html
+    - html_lint.py news/*.html
+    - jshint ./js/*.js --config .jshint
+    - stylelint "css/*.css" --config .stylelintrc.json
```
leave final blank line (configure your editor to do so automatically)

nemesisdesign commented on css/style.css:
```diff
@@ -150,48 +177,62 @@ a {
 }
+
 body.content .buttons { margin-bottom: -1em }
+
```
is it really necessary to have these blank lines? can you configure the linter to avoid requiring this?

cking100 replied:
@nemesisdesign Definitely, i just thought it makes the css more readable and clear.. And some files were using a line empty before next rule and some were not so to make each of them similar. Do lemme know if i should undo this

nemesisdesign replied:
I like to have compact code, if you feel lines are too close to one another you should simply increase the `line-height` property of your editor, I really prefer if you don't add all these blank lines because they just make the code longer for me to read

nemesisdesign commented on gsoc/ideas-2018.html:
```diff
@@ -59,7 +59,7 @@ <h1>Google Summer of Code 2018 Idea List</h1>
         </p>
         <p class="center">
           <a class="ui big inverted red button"
-             href="http://openwisp.io/docs/developer/google-summer-of-code.html">
```
I think this is not good because if we move the site to HTTPS all links will not work because not every link is available via HTTPs. This applies for all changes similar to this one. Can you find a way to avoid doing this? If necessary, use a different linter as mentioned before.

cking100 replied:
That is definitely not the correct way of doing this. I was just trying to avoid warnings to see if it works but now i will try a different linter as this one doesn't allow a customisable configuration file.

cking100 commented:
@nemesisdesign Definately it is not the correct choice of linter. It had scratching my head for a couple of days but thanks for informating if not, would still try to implement this for nothing and No issues i will try to implement using a different linter and get back to you asap :)

cking100 commented:
@nemesisdesign I have addressed your feedback just that `empty line before next rule` in css need a feedback. Please have a look when you are free thanks :)

atb00ker commented on .jshint:
```diff
@@ -1,7 +1,28 @@
+  "-W008": false,
+  "-W054": false,
+  "-W004": false
+}
```
Add an empty line at the end of the file. (You can set your editor to do this for you)

atb00ker commented on .stylelintignore:
```diff
@@ -0,0 +1 @@
+css/semantic.min.css
```
Add an empty line at the end of the file.  (You can set your editor to do this for you)

atb00ker commented on .stylelintrc.json:
```diff
@@ -0,0 +1,19 @@
+      } ],
+      "unit-whitelist": ["em", "rem", "%", "s", "px", "vh"]
+    }
+  }
```
Add an empty line at the end of the file.  (You can set your editor to do this for you)

atb00ker commented on .travis.yml:
```diff
@@ -0,0 +1,16 @@
+  - htmlhint **/*.html *.html --config htmlhint.conf
+  - jshint ./js/*.js --config .jshint
+  - stylelint "css/*.css" --config .stylelintrc.json
+  
```
Remove blank spaces here! :smile:

atb00ker commented on css/ie.css:
```diff
@@ -1 +1 @@
-.box {behavior: url(js/PIE.htc); position:relative;}
\ No newline at end of file
+.box {behavior: url(js/PIE.htc); position: relative;}
```
Add an empty line at the end of the file. (You can set your editor to do this for you)

atb00ker commented on htmlhint.conf:
```diff
@@ -0,0 +1,25 @@
+  "href-abs-or-rel": false,
+  "attr-unsafe-chars": true,
+  "head-script-disabled": true
+}
```
Add an empty line at the end of the file. (You can set your editor to do this for you)

cking100 commented:
@atb00ker I have addressed your feedback thanks for your feedback. Do lemme know for any futher suggestion :)

nemesisdesign commented:
great :+1:, keep in mind all the suggestions we have given you for the next patches
